11_structures/main.go

```diff
@@ -69,58 +73,522 @@ instances inside the function.
 
 E.g.
 
-    type square struct { ... }
-
-    func (s square) area() float64 { ... }
+	type square struct { ... }
 
-    func main() {
-        s := square{side: 10}
-        fmt.Println(s.area()) // 100
-    }
+	func (s square) area() float64 { ... }
 
+	func main() {
+	    s := square{side: 10}
+	    fmt.Println(s.area()) // 100
+	}
 */
 
+// TODO: Create a struct named `triangle` with fields `side1`, `side2` and
+// `side3` of type `float64`.
+type triangle struct {
+	side1 float64
+	side2 float64
+	side3 float64
+}
+
+// TODO: Create a function on the structure `area` that returns a `float64`.
+type area struct {
+	side float64
+}
+
+func (a area) areaSize() float64 {
+	return a.side * a.side
+}
+
+// TODO: Adapt your code from 02_boolean to check the triangle type. The
+// function should return a string with the type of the triangle.
+func (t triangle) triangleType() string {
+	if t.sumLengthSmallerLongestSide() {
+		return "invalid triangle values"
+	}
+	// any side <=0 return error
+	if t.side1 <= 0 || t.side2 <= 0 || t.side3 <= 0 {
+		return "invalid triangle values"
+	}
+	if t.side1 == t.side2 && t.side1 == t.side3 {
+		return "equilateral"
+	}
+	if t.side1 == t.side2 || t.side1 == t.side3 ||
+		t.side2 == t.side3 {
+		return "isosceles"
+	}
+	if t.side1 != t.side2 || t.side1 != t.side3 ||
+		t.side2 != t.side3 {
+		return "scalene"
+	}
+	return "Error 404: Triangle logic not yet found"
+}
+
+func (t triangle) longestSide() float64 {
+	if t.side1 < t.side2 && t.side2 < t.side3 {
+		return t.side3
+	}
+	if t.side1 < t.side3 && t.side3 < t.side2 {
+		return t.side2
+	}
+	return t.side1
+}
+
+func (t triangle) sumTwoShortestSides() float64 {
+	if t.side1 < t.side2 && t.side2 < t.side3 {
+		return t.side1 + t.side2
+	}
+	if t.side1 < t.side3 && t.side3 < t.side2 {
+		return t.side1 + t.side3
+	}
+	return t.side2 + t.side3
```

Comment:
Just for the fun of it, can you find a way to express this without doing any comparison and by using `t.longuestSide()` instead?

Reply:
Like this? Thought of doing a loop, but I think this is the simplest I can come up with.

Reply:
yup, that's exactly what I had in mind :)